lms/add-stripe-subscription-id

anathomical opened the pull request:
## WHAT
Add migration to create Subscription.stripe_subscription_id column
## WHY
We have historically been using `stripe_invoice_id` as a proxy for a bunch of Stripe-related values.  (It is, for instance, how we determine if someone paid for their subscription with a credit card.)  However, we're about to start using Stripe invoices in an additional way that isn't compatible with a bunch of these assumptions (specifically, we'll allow for the use of invoices without credit card payments).  In order to facilitate this logic, we want to begin storing a different value that we think represents a better proxy for our assumptions and we've settled in the Stripe `subscription_id` value.
## HOW
Just add a migration to create this column in the table.  There will be follow-up PRs to begin using this value once it is available.

### Notion Card Links
https://www.notion.so/quill/Modify-Stripe-subscription-storage-to-use-stripe_subscription_id-in-addition-to-stripe_invoice_id-64273f4887a24ccd9b81d2fbc983ef3e

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  No test changes, but factory updated
Have you deployed to Staging? | Deploying now
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

anathomical committed:
Add migration to create Subscription.stripe_subscription_id column

brendanshean reviewed (approved):
Looks good to me.

You could add a validation in the model file similar to stripe_invoice_id:
`validates :stripe_subscription_id, allow_blank: true, stripe_uid: { prefix: :sub }`

anathomical committed:
Update db/structure.sql with a migration reversion from an old branch

anathomical committed:
Tweak the Stripe webhooks to reference stripe_invoice_id

Since we want to differentiate between Subscriptions related to Stripe credit card transactions, and those with other types of Stripe invoices

anathomical committed:
Fix linting errors

anathomical committed:
Fix linting error

anathomical committed:
More linting

anathomical committed:
Clean up some specs

anathomical committed:
Merge branch 'develop' of github.com:empirical-org/Empirical-Core into lms/add-stripe-subscription-id

anathomical committed:
Add rake task to set stripe_subscription_id values on Subscription models

When there's a stripe_invoice_id

anathomical committed:
Add a validation based on code review

anathomical committed:
Merge branch 'develop' of github.com:empirical-org/Empirical-Core into lms/add-stripe-subscription-id

brendanshean reviewed (approved):
Nice work on this!
